src/threading/MsgThread.cc

```diff
@@ -339,6 +353,8 @@ void MsgThread::SendOut(BasicOutputMessage* msg, bool force)
 	queue_out.Put(msg);
 
 	++cnt_sent_out;
+
+	flare.Fire();
```

Comment:
Just to check - is it ok to fire the flare here?

Flare.h says something about flares not being thread safe. (It also says something about this being ok when calls are mutually exclusive - and it is just a pipe that is being read/written to, so I think it is ok - but I am not 100% sure after reading the comments).

Reply:
Looking at the comment, I think the concern is if you make a copy of a flare and then `Fire()` on multiple threads. This could be a problem because of the looping in `Fire()`. Since we're just firing on one thread exclusively and extinguishing on one thread exclusively, it's probably safe.

I say probably because I'm also not 100% sure.

Reply:
Could also be the comment is preserved from a time when they contained more state than just the pipe FDs (e.g. maybe stored a boolean value indicating emptiness just to avoid doing unnecessary read() in some cases).

Reading the code now, the only issue I can think of were if someone structures the consumer thread's logic like "Process all pending tasks, then Extinguish" since the producer thread can concurrently write a byte to the pipe that gets lost without performing the intended duty of signal'ing a new pending task.  If the consumer thread does "Extinguish then Process pending tasks", that should be fine.

Reply:
>  If the consumer thread does "Extinguish then Process pending tasks", that should be fine.

Which is what the thread code is doing 👍